### PR TITLE
Always enable force_config_drive for ZVM

### PR DESCRIPTION
**Pull request by mmnelemane**

### CODE REVIEW

**Comment by vuntz** [comment]
Do the options need to be set only on the compute node, or are they needed on the controller?

**Comment by dirkmueller** [comment]
+1

**Comment by dirkmueller** [comment]
@vuntz I only tested it on all nodes (e.g. by changing the crowbar attribute), which works, but thinking about it I guess only doing compute node should be fine. it is the local configuration only that the driver is reading anyway.

**Comment by mmnelemane** [comment]
https://github.com/crowbar/crowbar-openstack/pull/458 should do the backport

File: chef/cookbooks/nova/templates/default/nova.conf.erb [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ linuxnet_interface_driver = ""
 
 # Whether to attempt to inject network setup into guest (boolean value)
 #flat_injected = false
-<% if @force_config_drive %>
+<% if @force_config_drive or @libvirt_type.eql?('zvm') %>
```
Comment (vuntz) [inline chef/cookbooks/nova/templates/default/nova.conf.erb]: Our guidelines say to use `||` instead of `or`, but minor detail

**Comment by vuntz** [comment]
+1. Would be nice to fix the minor detail before merging, though

**Comment by mmnelemane** [comment]
@vuntz: fixed

**Pull request merged**